First implementation of Register Use case #2

sebastien-deren opened the pull request:
It will need a bit of work to clean the mess I have created 
But our use case is working as intended.

sebastien-deren committed:
First implementation of Register Use case #2
It will need a bit of work to clean the mess I have created
But our use case is working as intended.

sebastien-deren committed:
First implementation of Register Use case #2
It will need a bit of work to clean the mess I have created
But our use case is working as intended.

Lionelsan95 commented on src/Controller/DumpDataController.php:
```diff
@@ -0,0 +1,26 @@
+    #[Route('/dump/data', name: 'app_dump_data')]
+    public function index(UserRepository $userRepository, EntityManagerInterface $manager): Response
+    {
+        $users = $userRepository->findAll();
+        foreach ($users as $user){
+            $userRepository->remove($user);
+            $manager->flush();
+        }
```
@sebastien-deren logique à mettre dans un service. Ainsi, on peut l’appeler de n'importe où tant qu'on a les bons params.

sebastien-deren replied:
Je le sais bien, cette classe est uniquement une classe de test/debug. je la supprimerais avant de merge

Lionelsan95 commented on src/Service/VerifyEmailHelper.php:
```diff
@@ -0,0 +1,32 @@
+         return hash('md5',$userId.$username);
+    }
+
+}
```
@sebastien-deren quel est le but de cette classe ?

sebastien-deren replied:
Elle me servais a créer l'url de verification (et éventuellement d'autre données a envoyer).
Au final comme je n'ai besoin que d'une URl j'ai supprimé cette classe

Lionelsan95 commented on src/Controller/RegistrationController.php:
```diff
@@ -0,0 +1,95 @@
+            // generate a signed url and email it to the user
+            $this->emailVerifier->sendEmailConfirmation('app_verify_email', $user,
+                (new TemplatedEmail())
+                    ->from(new Address('mailer@snowtricks.com', 'Snowtricks mailer'))
```
mailer@snowtricks penses-tu qu'il doit être mis en dur ici ? 

regarde : https://symfony.com/doc/current/service_container.html#binding-arguments-by-name-or-type

sebastien-deren replied:
Merci du lien, c'est corrigé et bien noté !

Lionelsan95 commented on src/Repository/UserRepository.php:
```diff
@@ -19,11 +20,22 @@
 
+    public function registerNewUser(User $user,string $plainPassword="", bool $flush = false): void
+    {
+        if(!empty($plainPassword)){
```
Quand le password est empty, tu créais le compte ainsi ?

sebastien-deren replied:
J'ai réecrit cela, le Repository n'a pas besoin de savoir si on créer un nouvel utilisateur ou si on l'update. Et n'a pas a set le mot de passe.

Lionelsan95 commented on src/Security/EmailVerifier.php:
```diff
@@ -0,0 +1,60 @@
+    {
+
+        $email = (new TemplatedEmail())
+            ->from(new Address('mailer@snowtricks.com', 'Snowtricks mailer'))
```
encore ce mail en constante

Lionelsan95 commented on src/Security/EmailVerifier.php:
```diff
@@ -0,0 +1,60 @@
+    /**
+     * @throws VerifyEmailExceptionInterface
+     */
+    public function handleEmailConfirmation(Request $request, UserInterface $user): void
```
pas à mettre ici
les seules classes qui gèrent les requêtes http sont les controller

sebastien-deren committed:
Creation of our entity and repository (#19)

Creation of our 4 entity and repoistories
Creation of our first migration
We have succesfully implemented #1 & #3 Issues

sebastien-deren committed:
Cleaning and modification base on review
Suppression of useless classes
Creation of Interfaces for SOLID reason
(dependency inversion & Interface segregation)
/!\ the register service have more than one Responsibility (set pw, register in DB, send mail & verify mail)
Might break down in three classes ?

sebastien-deren committed:
Cleaning and modification base on review
Suppression of useless classes
Creation of Interfaces for SOLID reason
(dependency inversion & Interface segregation)
/!\ the register service have more than one Responsibility (set pw, register in DB, send mail & verify mail)
Might break down in three classes ?

sebastien-deren committed:
Better separation of interface
DI with precise interface who implement only the necessary methods.
Seting only in The repository and with the Form(FormType)

sebastien-deren committed:
Better separation of interface
DI with precise interface who implement only the necessary methods.
Seting only in The repository and with the Form(FormType)

Lionelsan95 commented on src/Controller/DumpDataController.php:
```diff
@@ -0,0 +1,30 @@
+
+class DumpDataController extends AbstractController
+{
+    #[Route('/dump/data/{db}', name: 'app_dump_data')]
+    public function index(UserRepository $userRepository, EntityManagerInterface $manager, string $db = null): Response
+    {
+        if (null === $db) {
```
@sebastien-deren $db est-il vraiment utilisé ?

sebastien-deren replied:
Controller en attente de fixtures, il sera supprimé prochainement

Lionelsan95 commented on src/Controller/RegistrationController.php:
```diff
@@ -0,0 +1,74 @@
+        // @TODO Change the redirect on success and handle or remove the flash message in your templates
+        $this->addFlash('success', 'Your email address has been verified.');
+
+        return $this->redirectToRoute('app_register');
```
@sebastien-deren cette portion de code risque ne jamais être exécutée

sebastien-deren replied:
Corrigé!

Lionelsan95 commented on src/Entity/Interface/UserRegisterInterface.php:
```diff
@@ -0,0 +1,9 @@
+
+namespace App\Entity\Interface;
+
+interface UserRegisterInterface
```
@sebastien-deren nécessaire ?

sebastien-deren replied:
Corrigé!

Lionelsan95 commented on src/Entity/Interface/VerifiableUserInterface.php:
```diff
@@ -0,0 +1,9 @@
+
+namespace App\Entity\Interface;
+
+interface VerifiableUserInterface
```
@sebastien-deren nécessaire ?

Lionelsan95 commented on src/Repository/UserRepository.php:
```diff
@@ -55,6 +57,11 @@ public function upgradePassword(PasswordAuthenticatedUserInterface $user, string
 
         $this->save($user, true);
     }
+public function verifyUser(VerifiableUserInterface $user, bool $verified): void
```
@sebastien-deren nom pas très parlant. 

j'ai l'impression que ça vérifie et dé-vérifie un user

sebastien-deren replied:
Corrigé!

Lionelsan95 commented on src/Service/Interface/ConfirmEmailInterface.php:
```diff
@@ -0,0 +1,10 @@
+
+use App\Entity\Interface\UserRegisterInterface;
+
+interface ConfirmEmailInterface
```
@sebastien-deren nécessaire ?

Lionelsan95 commented on src/Service/RegisterService.php:
```diff
@@ -0,0 +1,42 @@
+
+    public function verifyEmail(UserRegisterInterface $user, string $token): void
+    {
+        if (!hash_equals($token, hash('md5', $user->getId() . $user->getUsername()))) {
```
@sebastien-deren voir remarque autre PRs. il faut créer une méthode interne à la classe User

sebastien-deren replied:
Corrigé!

Lionelsan95 reviewed (commented):
waiting for update and/or modifications to test functionnally.

sebastien-deren committed:
Correction based upon mentor's review
Adding a hash method in User
deleting useless interfaces

sebastien-deren committed:
Correction based upon mentor's review
Adding a hash method in User
deleting useless interfaces

sebastien-deren committed:
Really basic controller and view, just a page to redirect into (#24)

Adding navigation to our base template (not designed just a list to navigate)

sebastien-deren committed:
Really basic controller and view, just a page to redirect into (#24)

Adding navigation to our base template (not designed just a list to navigate)

sebastien-deren committed:
using bootstrap to make a proper register page

Lionelsan95 commented:
Validé, fonctionnel après tests.

sebastien-deren committed:
adding a second field to confirm password

sebastien-deren committed:
Feature/implement login #4 (#23)

* Login implementation
Using make:auth
Looking for change

* Login implementation
Using make:auth
Looking for change

* correcting typo

* Really basic controller and view, just a page to redirect into (#24)

Adding navigation to our base template (not designed just a list to navigate)

* using bootstrap to make a proper login page

sebastien-deren committed:
correcting conflicting files

sebastien-deren committed:
correcting conflicting files

sebastien-deren committed:
Merge branch 'feature/Implement_Register#2' of https://github.com/sebastien-deren/Projet_6_snowtricks into feature/Implement_Register#2
